### DISCUSSION TITLE
BPSK modulation with LR11XX radios

### DISCUSSION THREAD

**Discussion by lyusupov** [jgromes/RadioLib#1325] [General]
[**CHANGELOG**](https://github.com/Lora-net/radio_firmware_images/blob/master/lr1110/transceiver/CHANGELOG.md) of the Semtech firmware says that:

-----------------------------
### Added

- BPSK modulation
-----------------------------------

<br>

1) are there any plans to add this modulation type feature into the RadioLib ?
2) has anybody seen a datasheet from Semtech on how operate BPSK with their radios ?

**Comment by jgromes** [comment]
Based on what I understood from their datasheets/user manuals, this is needed for SigFox support. However, information on how to operate in this mode is mostly not there, so further information from anybody that has it is most certainly welcome.

**Reply by lyusupov** [reply]
> However, information on how to operate in this mode is mostly not there

@jgromes 
Sigfox feature for SX126x and LR11xx is explained in this Semtech AppNote https://semtech.my.salesforce.com/sfc/p/E0000000JelG/a/RQ000001DuF8/tf5j58cShJ5vqSWMa_XciFcYb_zTeTU7i3S6q22wnmM

There are references in the AN onto Semtech GitHub repo with BPSK transmit API/examples/source code.

**Reply by jgromes** [reply]
Thank you for pointing me to that app note - with the information from that I was able to get an SX1262 to transmit BPSK. However, it seems that this feature is limited similarly to LR-FHSS:

1. SX126x and LR11x0 can only transmit, not receive.
2. All packet construction (preamble, sync word, CRC etc.) has to be done in software.
3. It is not clear what the actual limitations are. The application note claims that only 100 and 600 bps rates are supported, but the SX1262 itself seems capable of also transmitting higher rates, like 1.2 kbps. Similarly with pulse shaping, I was able to turn it off completely, but the docs/repos seems to suggest only BT=0.7 should be used. It is unclear whether these "limits" are only there because of SigFox.

I think support for BPSK on its own doesn't make much sense because it can only be transmitted. It makes the most sense if we were to implement SigFox on top of it. However, I'm not sure that's worth it. For starters, I don't know anything about SigFox, so it's hard to estimate how much effort would that take and whether the user base is wide enough to justify this addition. Coverage of SigFox is also quite limited where I live, which will make testing complicated. And finally, it all seems to be subscription-based, unlike LoRaWAN. Also, no radio currently supported by RadioLib supports BPSK.

So unless we have a convincing use-case, I will file this one under "hey wouldn't it be nice if ..."

**Reply by lyusupov** [reply]
> So unless we have a convincing use-case, I will file this one under "hey wouldn't it be nice if ..."

[SX126x] Add experimental BPSK support https://github.com/jgromes/RadioLib/commit/1de98370ebfd2c17b39a7bb2bf889766850fc2a9

**Reply by jgromes** [reply]
I admit I got a little bit carried away while going through the recent releases of Semtech drivers - you got me there ;) However, the upcoming LR2021 will include BPSK suport as well, and having that and LR11xx makes it a bit more convincing to implement at least a basic Tx-only BPSK support.
